### PR TITLE
Master small fixes

### PR DESCRIPTION
**Pull request by mingwandroid**
Eh, the cran change isn't that small thinking about it, the other one should catch the problem (or similar ones) that we had with setuptools_scm.

### CODE REVIEW

**Comment by mcg1969** [comment]
Can you briefly summarize what the functional changes are to the CRAN skeleton? They look potentially very useful but some I don't quite understand.

**Comment by jjhelmus** [comment]
Would conda-verify be a better location for the dist-info check?  It already has some similar checks for egg, and pth files.

**Comment by mingwandroid** [comment]
I dislike putting simple checks like this into conda-verify because (e.g.) we didn't run it for years and running this check should not be optional.

**Comment by mingwandroid** [comment]
The binary releases for R seem a bit haphazard, i.e. you'll find packages where the latest (at the snapshot date) source bundle is 1.2 but the latest macOS binary is 1.1 while windows has 1.2, so I add an option for what to do in the case when you wanted binary repackaging but the correct version does not exist. The options are to use the older one (may not work, depending on whether it was actually rebuilt or not) or to compile from source in the normal way.

I also do lots of dylib fixups so that our compiler runtimes are used instead of those provided by MRO (because once we load those old ones we cannot offer any modern language features).

File: conda_build/skeletons/cran.py [hunk 1]
```diff
@@ -143,6 +148,33 @@
...
+  if [[ $target_platform == osx-64 ]]; then
+    pushd $PREFIX
+      for libdir in lib/R/lib lib/R/modules lib/R/library lib/R/bin/exec sysroot/usr/lib; do
+        pushd $libdir || exit 1
+          for SHARED_LIB in $(find . -type f -iname "*.dylib" -or -iname "*.so" -or -iname "R"); do
+            echo "fixing SHARED_LIB $SHARED_LIB"
+            install_name_tool -change /Library/Frameworks/R.framework/Versions/3.5.0-MRO/Resources/lib/libR.dylib "$PREFIX"/lib/R/lib/libR.dylib $SHARED_LIB || true
+            install_name_tool -change /Library/Frameworks/R.framework/Versions/3.5/Resources/lib/libR.dylib "$PREFIX"/lib/R/lib/libR.dylib $SHARED_LIB || true
+            install_name_tool -change /usr/local/clang4/lib/libomp.dylib "$PREFIX"/lib/libomp.dylib $SHARED_LIB || true
+            install_name_tool -change /usr/local/gfortran/lib/libgfortran.3.dylib "$PREFIX"/lib/libgfortran.3.dylib $SHARED_LIB || true
```
Comment (msarahan) [inline conda_build/skeletons/cran.py]: the libgfortran so version changes in recent GCC (it's on 5 in gcc 8.2).  We're still on gcc 4.8 for mac, so it doesn't matter here yet, but should we constrain libgfortran somehow in light of this?

Comment (mingwandroid) [reply conda_build/skeletons/cran.py]: If I need to pin libgfortran I'll do it in r-base only I think.

**Pull request merged**